addOns/domxss/src/main/java/org/zaproxy/zap/extension/domxss/DomXssScanRule.java

```diff
@@ -430,30 +455,54 @@ private void getHelper(WebDriverWrapper wrapper, String url, int retry) {
         return new ArrayList<>();
     }
 
+    private String getAlertDialogText() {
+        try {
+            org.openqa.selenium.Alert alertDialog = driver.getDriver().switchTo().alert();
+            if (alertDialog != null) {
```

Comment:
Can `alertDialog` be null? It seems that `alert()` either throws or returns the dialogue.